### PR TITLE
Introduce RankingEvaluator and test

### PR DESCRIPTION
**Pull request by almumill**
Introduces a first pass at a RankingEvaluator with an MRR metric for categorical atoms with a single category index, as well as a unit test for this evaluator and updated options.

### CODE REVIEW

File: psl-core/src/test/java/org/linqs/psl/evaluation/statistics/RankingEvaluatorTest.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+import org.linqs.psl.model.term.ConstantType;
+import org.linqs.psl.model.term.UniqueIntID;
+import org.linqs.psl.util.MathUtils;
+
+import org.linqs.psl.util.MathUtils;
+
+import org.junit.Before;
+import org.junit.Test;
+
+public class RankingEvaluatorTest extends EvaluatorTest<RankingEvaluator> {
```
Comment (eriq-augustine) [inline psl-core/src/test/java/org/linqs/psl/evaluation/statistics/RankingEvaluatorTest.java:43]: I think you should favor tests in the style of AUREvaluatorTest (since it is newer) rather than CategoricalEvaluatorTest.

It will take a little more work making an `init()` that works, but the idea is that you can then make many cases easily and test against a reference implementation.
For the AUREvaluatorTest, I just put in some test cases than they generated random numbers and tested against scikit-learn.
I even made it so that my script that generated the numbers and ran sklearn also output the Java code so I could just copy and paste.

**Review by eriq-augustine — CHANGES_REQUESTED** [review]
Good start.

Be careful, there are a lot a base style things here that you should have caught.
Remember that I will typically only make a comment once or twice even if it appears more.
